elixir/lib/samson_secret_puller.ex

```diff
@@ -0,0 +1,69 @@
+defmodule SamsonSecretPuller do
+  @moduledoc """
+  Wait for secrets to appear from the secrets sidecar and then build a `Keyword`
+  list of the secrets.
+  """
+  @folder "/secrets"
+  @timeout 60
+  @sleep_interval 0.1
+
+
+  @doc """
+  Says hello to the given `name`.
+
+  Returns `[key: "value"]`.
+
+  ## Examples
+
+      iex> SamsonSecretPuller.fetch_secrets!
+      [MYSQL_USER: "admin", MYSQL_PASS: "password"]
+
+  """
+  @spec fetch_secrets! :: Keyword.t
+  def fetch_secrets! do
+    case wait_for_secrets_to_appear do
+      {:ok, _}      -> read_secrets
+      {:error, msg} -> []
+    end
+  end
+
```

Comment:
everything defined with `defp` is private. So there is only one public method, `fetch_secrets!` and the rest are private.